tests/simulator_utils_test.py

```diff
@@ -30,18 +30,19 @@ def test_simulate_in_batches(
     simulate_in_batches(simulator, theta, batch_size)
 
 
-def test_inference_with_pilot_samples_many_samples():
-    """Test whether num_pilot_sims can be same as num_simulations_per_round."""
+def test_inference_with_exactly_sim_batch_size_sims_per_round():
```

Comment:
this test ([commit b04323](https://github.com/mackelab/sbi/commit/b04323902e9ea919eb1bc1c4e41efe1e678c67d6)) was supposed to test the case when num pilot equals equals num samples in first round such that all of them are recycled and no more simulations need to be run for the first round. 

We constructed this test to detect a bug in `simulate_in_batches` when num_samples=0. We then fixed this bug with [this commit](https://github.com/mackelab/sbi/commit/49ccf82374be7a8d3b9cb38dacc497595d1eed23). 

Once pilot samples are I think we dont need this test anymore. 

However, now the test is different and I am not sure what it is supposed to test.

Reply:
But there was no mention whatsoever of `num_pilot_sims` in the test.

I just gave it my best guess at interpreting the test. In the future it might be a good idea to link bug-fixing tests to the particular issue they are solving? 

I will remove it.